src/Network/Algorand/Client.hs

```diff
@@ -51,15 +51,18 @@ connectToNode
   -- ^ Host of the node.
   -> Network
   -- ^ Expected network (genesis id).
-  -> m (Network, AlgoNode)
+  -> m (Network, AlgoNode m)
 connectToNode host net = do
   manager <- newTlsManager
   env <- mkClientEnv manager <$> parseBaseUrl (T.unpack host)
   let
-    apiV2Client :: forall m' . MonadIO m' => NodeApi (AsClientT m')
-    apiV2Client = apiClient env
+    apiV2Client :: NodeApi (AsClientT m)
+    apiV2Client = apiClient @_ @m env
 
-  pure (net, AlgoNode apiV2Client)
+  Version{vGenesisId} <- _version $ apiV2Client
+  if vGenesisId == net then
```

Comment:
```
if ..
then ...
else ...
```
as I remember we have such a requirement at our style guide

Reply:
Yeah, but it looks a bit confusing. I change it to `match` case instead.